OpenAPITools/openapi-generator
# [dart] Cleanup and OAS3 support

* cleanup old docs across all dart generators
* regenerate all tests
* cleanup .gitignore
* minor formatting improvements in dart-dio
* unified integration-test layout for dart/dart-dio generators
* add generation and integration-tests for OAS3 petstore to dart/dart-dio
* add generation and integration-tests for OAS3 petstore-with-fake-api to dart/dart-dio

**Explicitly not adding the new integration tests for petstore-with-fake-api to the master POM. They do not work but having them allows for quickly iterating on open issues without breaking existing basic OAS2/OAS3 petstore examples. Instead they should be run manually until everything is fixed.**


### PR checklist
 
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

**TODO:**
- [x] Verify new inline objects are correct, they seem to be unused
- [x] Fix compile errors in dart regarding File/MultipartFile
- [x] Fix tests for POST/PUT operations that now return `200` with content
